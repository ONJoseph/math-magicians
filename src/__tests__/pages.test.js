import React from 'react';
import { render } from '@testing-library/react';
import Home from '../components/Home';
import Quote from '../components/Quote';

describe('Testing the Home and Quote pages render correctly', () => {
  it('Home page renders correctly', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Quote page renders correctly', () => {
    const { asFragment } = render(<Quote />);
    expect(asFragment()).toMatchSnapshot();
  });
});