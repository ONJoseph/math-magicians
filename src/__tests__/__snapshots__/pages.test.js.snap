// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Home and Quote pages render correctly Home page renders correctly 1`] = `
<DocumentFragment>
  <main
    class="home"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Mathematics, the science of structure, order, and relation that has evolved from elemental practices of counting, measuring, and describing the shapes of objects. It deals with logical reasoning and quantitative calculation, and its development has involved an increasing degree of idealization and abstraction of its subject matter. The study of mathematics as a "demonstrative discipline" began in the 6th century BC with the Pythagoreans, who coined the term "mathematics" from the ancient Greek μάθημα (mathema), meaning "subject of instruction". Since the 17th century, mathematics has been an indispensable adjunct to the physical sciences and technology, and in more recent times it has assumed a similar role in the quantitative aspects of the life sciences.
    </p>
    <p>
      In many cultures—under the stimulus of the needs of practical pursuits, such as commerce and agriculture—mathematics has developed far beyond basic counting. This growth has been greatest in societies complex enough to sustain these activities and to provide leisure for contemplation and the opportunity to build on the achievements of earlier mathematicians. At the end of the 19th century the International Congress of Mathematicians was founded and continues to spearhead advances in the field.
    </p>
  </main>
</DocumentFragment>
`;

exports[`Testing the Home and Quote pages render correctly Quote page renders correctly 1`] = `
<DocumentFragment>
  <div
    class="quote"
  >
    <h2
      style="text-align: center;"
    >
      Enjoy our random Math quotes!
    </h2>
    <p
      class="quote-line"
    >
      Mathematics is the queen of science, and arithmetic the queen of mathematics. — Carl Friedrich Gauss
    </p>
  </div>
</DocumentFragment>
`;
